app/io/pathfinder/controllers/ClusterController.java

```diff
@@ -0,0 +1,104 @@
+package io.pathfinder.controllers;
```

Comment:
General note on this file, and probably other files:

Why do these methods need to be public?